apps/coordinator/src/components/Wallet/TransactionsTab/FeeBumping/context/hooks.ts

```diff
@@ -0,0 +1,201 @@
+
+import { FeeBumpStatus, FeeBumpRecommendation } from "../types";
+import { extractUtxosForFeeBumping, validateTransactionInputs } from "../utils";
+import { TransactionDetails } from "@caravan/clients";
+import { useFeeBumpContext } from "./FeeBumpContext";
+// =============================================================================
+// HOOKS
+// =============================================================================
+
+/**
+ * Identifies the change output in a transaction by analyzing output addresses
+ * and wallet data
+ *
+ * This function uses multiple heuristics to identify which output is the change:
+ * 1. Matches against known wallet addresses
+ * 2. Checks BIP32 path patterns (change addresses use path m/1/*)
+ * 3. Position in outputs (change is often the last output)
+ *
+ * @param transaction - The transaction object
+ * @param walletState - The wallet state containing addresses
+ * @returns Index of the change output or undefined if not found
+ *
+ * @see https://en.bitcoin.it/wiki/Privacy#Change_address_detection
+ */
+export const useChangeOutputIndex = (
+  transaction?: TransactionDetails,
+): number | undefined => {
+  const changeAddresses = useSelector(getChangeAddresses);
+  const walletAddresses = useSelector(getWalletAddresses);
+
+  if (!transaction) return undefined;
+  if (!transaction.vout?.length) return undefined;
+
+  const changeAddressesSet = new Set(changeAddresses);
+  const walletAddressesSet = new Set(walletAddresses);
+
+  // 1) First look for any explicit change‑address hits
+  for (let i = 0; i < transaction.vout.length; i++) {
+    const addr = transaction.vout[i].scriptPubkeyAddress;
+    if (addr && changeAddressesSet.has(addr)) {
+      return i;
+    }
+  }
+
+  // 2) : Check if any output goes to a known wallet address
+  // This is less reliable but can help identify change when the exact
+  // change address isn't recognized
+  for (let i = 0; i < transaction.vout.length; i++) {
+    const addr = transaction.vout[i].scriptPubkeyAddress;
+    if (addr && walletAddressesSet.has(addr)) {
+      return i;
+    }
+  }
+
+  return undefined;
+};
+
+export const useGetAvailableUtxos = () => {
+  const {
+    state: { transaction },
+  } = useFeeBumpContext();
+  const { utxos: pendingUtxos } = usePendingUtxos(transaction?.txid || "");
+  const walletUtxos = useWalletUtxos();
+
+  // Memoize the combined UTXOs so it only recalculates when dependencies change
+  return useMemo(
+    () => extractUtxosForFeeBumping(pendingUtxos || [], walletUtxos || []),
+    [pendingUtxos, walletUtxos, transaction?.txid],
+  );
+};
+
+export const useGetGlobalXpubs = () => {
+  const getGlobalXpubs = useSelector(getExtendedPublicKeyImporters); // same for the whole wallet
+  return Object.values(getGlobalXpubs).map((item: any) => ({
+    masterFingerprint: item.rootXfp,
+    path: item.bip32Path,
+    xpub: item.extendedPublicKey,
+  }));
+};
+
+export const useAnalyzeTransaction = () => {
```

Comment:
i collapsed the other analyze transaction function into this hook so we have everything in one place. 

Since this only really needs to be run once when the transaction gets loaded up, this should work fine as a hook like this. 

What's interesting is that nothing is returned from this. It basically just watches for changes in the state, runs the analysis and dispatches changes. I think though after looking at this, if we don't care about changing the status, then this can just return the recommendations and those can be used to return fee bump recommendations based on a TransactionDetails object being passed in.